packages/contractkit/src/network-utils/static-node-utils.ts

```diff
@@ -1,22 +1,155 @@
+  const rangeCheck = (x: number, range: Range | undefined) =>
+    range === undefined ||
+    (range.min <= range.max ? range.min <= x && x < range.max : range.min <= x || x < range.max)
+
+  return (
+    rangeCheck(coords.lattitude, region.lattitude) && rangeCheck(coords.longitude, region.longitude)
+  )
+}
+
+/**
+ * Region names for groups of static nodes deployed around the world for each network.
+ * Used as a suffix to the blob name when fetching to get region specific nodes.
+ */
+const StaticNodeRegions: { [network: string]: Region[] } = {
+  mainnet: [
+    {
+      name: 'gcp-asia-east1',
+      longitude: { min: 60, max: -170 },
+    },
+    {
+      name: 'gcp-europe-west1',
+      longitude: { min: -25, max: 60 },
+    },
+    {
+      name: 'gcp-southamerica-east1',
+      lattitude: { min: -90, max: 0 },
+      longitude: { min: -170, max: -25 },
+    },
+    {
+      name: 'gcp-us-east1',
+      lattitude: { min: 0, max: 90 },
+      longitude: { min: -105, max: -25 },
+    },
+    {
+      name: 'gcp-us-west1',
+      lattitude: { min: 0, max: 90 },
+      longitude: { min: -170, max: -105 },
+    },
+  ],
+  // Alias for mainnet
+  rc1: [
+    {
+      name: 'gcp-asia-east1',
+      longitude: { min: 60, max: -170 },
+    },
+    {
+      name: 'gcp-europe-west1',
+      longitude: { min: -25, max: 60 },
+    },
+    {
+      name: 'gcp-southamerica-east1',
+      lattitude: { min: -90, max: 0 },
+      longitude: { min: -170, max: -25 },
+    },
+    {
+      name: 'gcp-us-east1',
+      lattitude: { min: 0, max: 90 },
+      longitude: { min: -105, max: -25 },
+    },
+    {
+      name: 'gcp-us-west1',
+      lattitude: { min: 0, max: 90 },
+      longitude: { min: -170, max: -105 },
+    },
+  ],
+}
+
 export class StaticNodeUtils {
   static getStaticNodesGoogleStorageBucketName(): string {
     return StaticNodesGoogleStorageBucketName
   }
 
   /**
-   * Fetches the static nodes (as JSON data) from Google Storage.
+   * Resolves the best region to use for static node connections.
+   * @param networkName Name of the network to get a region for.
+   * @remarks This method currently uses the interpreter's timezone and the
+   * IANA timezone database to establish what region of the world the client is
+   * in, then map that to a static list of static node clusters run by cLabs.
+   * If the timezone is not set according to the user's location, this method
+   * may route them to suboptimal set of static nodes. The resolution method
+   * may be replaced in the future.
+   */
+  static getStaticNodeRegion(networkName: string): string {
+    // Get the lattitude and longitude of the timezone locations.
+    // Note: This is the location of the city that the user has the timzone set to.
+    const tz = timezone()
+    const coords = tz.coordinates
+    if (coords === undefined) {
+      debug('Could not resolve region from timezone %s', tz.name)
+      return '' // Use the default region of static nodes
```

Comment:
this is a good idea-- i just realized that rn the file just called `rc1` (or `mainnet`) contains the enodes for the VM-based nodes. I assume we'd like to no longer have to maintain those a few months out from now (some time to account for old valora versions). Any take on which enodes we should put in the default one? Could be a basket of enodes (this would be much more annoying from an implementation perspective though) or simply from a specific context/region

Reply:
Because we have them deployed, using a basket of nodes makes the most sense to me. My understanding of LES as it stands is that it does not prioritize sending requests to nodes by latency, so we may want to exclude `southamerica-east1` from that basket, because is not as well connected to the rest of the world and users in Africa or the Phillipines would experience poor performance if their client routes requests there. If we want to use regions which will be workable for all users, `us-west1` and `us-east1` are the best connected globally. (On Google's network, all roads lead to Atlanta, and more recently, to Iowa and Oregon. There is also only a low bandwidth link between India and the Middle East, so traffic from travelling from Asia to Europe is often forced to go through America) https://cloud.google.com/about/locations

Reply:
Hmm okay, I will keep this in mind. Maybe there can be an env variable that lists the contexts that the default static nodes will consist of, like
```
FORNO_STATIC_NODE_DEFAULT_CONTEXTS=gcp-us-west1,gcp-us-east1
```
and whenever there's an update to one of those contexts, the static node file is reuploaded